sentry/src/test/java/io/sentry/ScopesTest.kt

```diff
@@ -0,0 +1,2186 @@
+    @Test
+    fun `when setFingerprint is called on disabled client, do nothing`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+        scopes.close()
+
+        val fingerprint = listOf("abc")
+        scopes.setFingerprint(fingerprint)
+        assertEquals(0, scope?.fingerprint?.count())
+    }
+
+    @Test
+    fun `when setFingerprint is called with null parameter, do nothing`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+
+        scopes.callMethod("setFingerprint", List::class.java, null)
+        assertEquals(0, scope?.fingerprint?.count())
+    }
+
+    @Test
+    fun `when setFingerprint is called, fingerprint is set`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+
+        val fingerprint = listOf("abc")
+        scopes.setFingerprint(fingerprint)
+        assertEquals(1, scope?.fingerprint?.count())
+    }
+    //endregion
+
+    //region clearBreadcrumbs tests
+    @Test
+    fun `when clearBreadcrumbs is called on disabled client, do nothing`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+        scopes.addBreadcrumb(Breadcrumb())
+        assertEquals(1, scope?.breadcrumbs?.count())
+
+        scopes.close()
+
+        assertEquals(0, scope?.breadcrumbs?.count())
+    }
+
+    @Test
+    fun `when clearBreadcrumbs is called, clear breadcrumbs`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+
+        scopes.addBreadcrumb(Breadcrumb())
+        assertEquals(1, scope?.breadcrumbs?.count())
+        scopes.clearBreadcrumbs()
+        assertEquals(0, scope?.breadcrumbs?.count())
+    }
+    //endregion
+
+    //region setTag tests
+    @Test
+    fun `when setTag is called on disabled client, do nothing`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+        scopes.close()
+
+        scopes.setTag("test", "test")
+        assertEquals(0, scope?.tags?.count())
+    }
+
+    @Test
+    fun `when setTag is called with null parameters, do nothing`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
+            scope = it
+        }
+
+        scopes.callMethod("setTag", parameterTypes = arrayOf(String::class.java, String::class.java), null, null)
+        assertEquals(0, scope?.tags?.count())
+    }
+
+    @Test
+    fun `when setTag is called, tag is set`() {
+        val scopes = generateScopes()
+        var scope: IScope? = null
+        scopes.configureScope {
```

Comment:
can't you use the getters now instead? e.g.
```suggestion
        val scope = scopes.getIsolationScope()
```

Reply:
While the getter is now still in place, I'm still thinking about duplicating this test for Android and it defaulting to current scope. Not sure that adds much value. Changing from `configureScope` to `getIsolationScope` would make it harder to switch tests. I'd say we revisit this at a later time.